File: Runtime/InitialConditionsDistribution.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

''' <summary>
''' Initial Conditions Distribution
''' </summary>
''' <remarks></remarks>
Friend Class InitialConditionsDistribution

    Private m_StratumId As Integer
    Private m_Iteration As Nullable(Of Integer)
    Private m_SecondaryStratumId As Nullable(Of Integer)
    Private m_TertiaryStratumId As Nullable(Of Integer)
    Private m_StateClassId As Integer
    Private m_AgeMin As Integer
    Private m_AgeMax As Integer
    Private m_RelativeAmount As Double

    Public Sub New(
        ByVal stratumId As Integer,
        ByVal iteration As Nullable(Of Integer),
        ByVal secondaryStratumId As Nullable(Of Integer),
        ByVal tertiaryStratumId As Nullable(Of Integer),
        ByVal stateClassId As Integer,
        ByVal ageMin As Integer,
        ByVal ageMax As Integer,
        ByVal relativeAmount As Double)

        Me.m_StratumId = stratumId
        Me.m_Iteration = iteration
        Me.m_SecondaryStratumId = secondaryStratumId
        Me.m_TertiaryStratumId = tertiaryStratumId
        Me.m_StateClassId = stateClassId
        Me.m_AgeMin = ageMin
        Me.m_AgeMax = ageMax
        Me.m_RelativeAmount = relativeAmount

    End Sub

    ''' <summary>
    ''' Gets the stratum Id
    ''' </summary>
    Public ReadOnly Property StratumId As Integer
        Get
            Return Me.m_StratumId
        End Get
    End Property

    ''' <summary>
    ''' Gets the Iteration
    ''' </summary>
    Public ReadOnly Property Iteration As Nullable(Of Integer)
        Get
            Return m_Iteration
        End Get
    End Property

    ''' <summary>
    ''' Gets the Secondary Stratum Id
    ''' </summary>
    Public ReadOnly Property SecondaryStratumId As Nullable(Of Integer)
        Get
            Return Me.m_SecondaryStratumId
        End Get
    End Property

    ''' <summary>
    ''' Gets the Tertiary Stratum Id
    ''' </summary>
    Public ReadOnly Property TertiaryStratumId As Nullable(Of Integer)
        Get
            Return Me.m_TertiaryStratumId
        End Get
    End Property

    ''' <summary>
    ''' StateClass Id for the cell
    ''' </summary>
    Public ReadOnly Property StateClassId As Integer
        Get
            Return Me.m_StateClassId
        End Get
    End Property

    ''' <summary>
    ''' Minimum age for the pathway
    ''' </summary>
    ''' <remarks></remarks>
    Public ReadOnly Property AgeMin As Integer
        Get
            Return Me.m_AgeMin
        End Get
    End Property

    ''' <summary>
    ''' Maximum age for the pathway
    ''' </summary>
    ''' <remarks></remarks>
    Public ReadOnly Property AgeMax As Integer
        Get
            Return Me.m_AgeMax
        End Get
    End Property

    ''' <summary>
    ''' Total amount (e.g. area) for the simulation
    ''' </summary>
    Public ReadOnly Property RelativeAmount As Double
        Get
            Return Me.m_RelativeAmount
        End Get
    End Property

End Class

File: src/Runtime/InitialConditionsDistribution.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2021 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

namespace SyncroSim.STSim
{
    internal class InitialConditionsDistribution
    {
        private int m_StratumId;
        private int? m_Iteration;
        private int? m_SecondaryStratumId;
        private int? m_TertiaryStratumId;
        private int m_StateClassId;
        private int m_AgeMin;
        private int m_AgeMax;
        private int? m_TSTGroupId;
        private int? m_TSTMin;
        private int? m_TSTMax;
        private double m_RelativeAmount;

        public InitialConditionsDistribution(
            int stratumId, 
            int? iteration, 
            int? secondaryStratumId, 
            int? tertiaryStratumId, 
            int stateClassId, 
            int ageMin, 
            int ageMax,
            int? tstGroupId,
            int? tstMin,
            int? tstMax,
            double relativeAmount)
        {
            this.m_StratumId = stratumId;
            this.m_Iteration = iteration;
            this.m_SecondaryStratumId = secondaryStratumId;
            this.m_TertiaryStratumId = tertiaryStratumId;
            this.m_StateClassId = stateClassId;
            this.m_AgeMin = ageMin;
            this.m_AgeMax = ageMax;
            this.m_TSTGroupId = tstGroupId;
            this.m_TSTMin = tstMin;
            this.m_TSTMax = tstMax;
            this.m_RelativeAmount = relativeAmount;
        }

        /// <summary>
        /// Gets the stratum Id
        /// </summary>
        public int StratumId
        {
            get
            {
                return this.m_StratumId;
            }
        }

        /// <summary>
        /// Gets the Iteration
        /// </summary>
        public int? Iteration
        {
            get
            {
                return m_Iteration;
            }
        }

        /// <summary>
        /// Gets the Secondary Stratum Id
        /// </summary>
        public int? SecondaryStratumId
        {
            get
            {
                return this.m_SecondaryStratumId;
            }
        }

        /// <summary>
        /// Gets the Tertiary Stratum Id
        /// </summary>
        public int? TertiaryStratumId
        {
            get
            {
                return this.m_TertiaryStratumId;
            }
        }

        /// <summary>
        /// StateClass Id for the cell
        /// </summary>
        public int StateClassId
        {
            get
            {
                return this.m_StateClassId;
            }
        }

        /// <summary>
        /// Minimum age for the pathway
        /// </summary>
        /// <remarks></remarks>
        public int AgeMin
        {
            get
            {
                return this.m_AgeMin;
            }
        }

        /// <summary>
        /// Maximum age for the pathway
        /// </summary>
        /// <remarks></remarks>
        public int AgeMax
        {
            get
            {
                return this.m_AgeMax;
            }
        }

        /// <summary>
        /// Gets the TST Transition Group ID
        /// </summary>
        public int? TSTGroupId
        {
            get
            {
                return this.m_TSTGroupId;
            }
        }

        /// <summary>
        /// Gets the TST Min
        /// </summary>
        public int? TSTMin
        {
            get
            {
                return this.m_TSTMin;
            }
        }

        /// <summary>
        /// Gets the TST Max
        /// </summary>
        public int? TSTMax
        {
            get
            {
                return this.m_TSTMax;
            }
        }

        /// <summary>
        /// Total amount (e.g. area) for the simulation
        /// </summary>
        public double RelativeAmount
        {
            get
            {
                return this.m_RelativeAmount;
            }
        }
    }
}
